src/SFA.DAS.EmployerIncentives.Commands/Persistence/AccountDomainRepository.cs

```diff
@@ -24,12 +26,19 @@ public async Task<Account> Find(long id)
         public Task Save(Account aggregate)
         {
             if (aggregate.IsNew)
-            {               
+            {
                 return _accountDataRepository.Add(aggregate.GetModel());
             }
 
             return _accountDataRepository.Update(aggregate.GetModel());
         }
+
+        public async Task<IEnumerable<Account>> GetByLegalEntityId(long legalEntityId)
```

Comment:
This shouldn't be here.  Use the existing AccountQueryRepository repository in the command handler to get the account Id for the legal entity in the command handler then use the IAccountDomainRepository.Find method.  The code already exists.